### DISCUSSION TITLE
May I deploy FastAPI using systemd directly?

### DISCUSSION THREAD

**Discussion by desjare** [fastapi/fastapi#7034] [Questions] [answered]
For some reasons, I cannot deploy FastAPI using docker. My application is working and I was asked to deploy it using systemd but I didn't find any example of this using FastAPI and I am not sure of the right approach:
* Should we use gunicorn (there is documentation about deploying gunicorn in systemd)?
* Can we just do it with FastAPI?
* Is there examples or documentation I missed? 

Thanks.

**Comment by includeamin** [comment, marked as answer]
Hi,
`Should we use gunicorn?`
- It's recommended to use gunicorn with uvicorn's worker on production. [link](https://www.uvicorn.org/#running-with-gunicorn)

`(there is documentation about deploying gunicorn in systemd)`
- not for FastApi but  [this](https://bartsimons.me/gunicorn-as-a-systemd-service/) may helps you and in your case, you can use `supervisor` to.

`Can we just do it with FastAPI?`
- what do you mean?

**Comment by desjare** [comment]
Thanks! 

Sorry for the confusion. I mean deploy using systemd without gunicorn.
I think I got what I needs Really appreciated!

**Comment by windson** [comment]
You may want to refer [this](https://www.tutlinks.com/deploy-fastapi-on-ubuntu-gunicorn-caddy-2/) tutorial, that help you Deploy FastAPI on Ubuntu and Serve using Caddy 2 Web Server

**Comment by desjare** [comment]
Thanks! We nailed need!

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:
